sphinx/domains/cpp/_symbol.py

```diff
@@ -935,6 +998,54 @@ def direct_lookup(self, key: LookupKey) -> Symbol:
             Symbol.debug_indent -= 2
         return s
 
+    def _resolve_alias_or_base_type(self) -> Generator[Symbol, None, None]:
+        resolved = self._resolve_type_alias()
+        if resolved is not None:
+            yield resolved
+        declaration = self.declaration
+        if declaration is None:
+            return
+        if declaration.objectType != "class":
+            return
+        for base in declaration.declaration.bases:
+            symbols, failReason = self.parent.find_name(
+                base.name, templateDecls=[], typ='any',
+                matchSelf=False,
+                recurseInAnon=True,
+                searchInSiblings=False)
+            if symbols:
+                yield symbols[0]
+
+    def _resolve_type_alias(self) -> Symbol | None:
+        """Resolves `self` to another symbol if it is a type alias."""
+        declaration = self.declaration
+        if declaration is None:
+            return None
+        if declaration.objectType != "type":
+            return None
+        nested_name: ASTNestedName
+        if (isinstance(declaration.declaration, ASTTypeUsing) and
+                declaration.declaration.type is not None):
+            trailing_type_spec = declaration.declaration.type.declSpecs.trailingTypeSpec
+            if not isinstance(trailing_type_spec, ASTTrailingTypeSpecName):
+                return None
+            nested_name = trailing_type_spec.name
+        elif isinstance(declaration.declaration, ASTType):
+            trailing_type_spec = declaration.declaration.declSpecs.trailingTypeSpec
+            if not isinstance(trailing_type_spec, ASTTrailingTypeSpecName):
+                return None
+            nested_name = trailing_type_spec.name
+        else:
+            return None
+        symbols, failReason = self.parent.find_name(
+            nested_name, templateDecls=[], typ='any',
```

Comment:
I'm not sure if ``any`` is technically correct here, but I don't think it matters too much right now (as long as it is not ``class``).